### ISSUE TITLE
train.py运行时全是0

### ISSUE THREAD

**Issue by irs442** [XiangZ-0/EVDI#14]
作者您好！我是一名正在学习的学生，冒昧想问一下您这个在pycharm里面运行的时候，别的都可以运行，但是Train.py这个模块运行出来的结果全都是0，像这种：
F:\1EVDI_python3.7\3.7explanation\Scripts\python.exe F:\1EVDI_python3.7\codes\Train.py Total number of training samples: 1 Begin training ./PreTrained/EVDI/...
---------------- Summary of Epoch ---------------- | L_B_S: 0.0000 | L_B_E: 0.0000 | L_S_E: 0.0000 | Total training loss: 0.000000 . Saving------------- Best trainLoss: 0.00000
---------------- Summary of Epoch ---------------- | L_B_S: 0.0000 | L_B_E: 0.0000 | L_S_E: 0.0000 | Total training loss: 0.000000 .
---------------- Summary of Epoch ----------------
| L_B_S: 0.0000 | L_B_E: 0.0000 | L_S_E: 0.0000 |
Total training loss: 0.000000 .
想请教一下是哪方面可能出问题了呢，就是感觉可能数据没有被这个文件捕捉到一样，希望能得到您的答复，谢谢！

**Comment by XiangZ-0** [comment]
你好，感谢对我们工作的关注！从你的结果里看好像你的训练集只有一个sample对吗？由于EVDI训练时默认batch_size=4（每次训练会读取4个sample），而且会抛弃最后一个不够4个sample的batch（具体可以看[这一行代码](https://github.com/XiangZ-0/EVDI/blob/main/codes/Train.py#L34)里drop_last=True），因此训练并没有进行。解决方法有两个：
1. 你可以在[这里](https://github.com/XiangZ-0/EVDI/blob/main/codes/Train.py#L162)设置batch_size=1，这样应该可以读取到你的数据，不过我猜这样训练应该很不稳定，效果可能不好。
2. 你可以下载比较大的数据集，比如我们另一个工作[GEM](https://github.com/XiangZ-0/GEM?tab=readme-ov-file#download-model-and-data)里的数据集（如Ev-REDS）进行训练，效果应该会更好一些。

希望对你的工作有帮助 :)

**Comment by irs442** [comment]
是的谢谢您，方法确实是奏效的，而且也确实像您说的一样如果用解决办法1效果不是很好，用2会好一些，再次谢谢您！
